packages/popper/src/index.vue

```diff
@@ -0,0 +1,466 @@
+      default: 'bottom' as Placement,
+    },
+    popperClass: {
+      type: String,
+      default: '',
+    },
+    referrer: {
+      type: HTMLElement as PropType<Nullable<HTMLElement>>,
+      default: null as Nullable<HTMLElement>,
+    },
+    showArrow: {
+      type: Boolean,
+      default: true,
+    },
+    strategy: {
+      type: String as PropType<PositioningStrategy>,
+      default: 'fixed' as PositioningStrategy,
+    },
+    transition: {
+      type: String,
+      default: 'el-fade-in-linear',
+    },
+    tabIndex: {
+      type: String,
+      default: '0',
+    },
+    value: {
+      type: Boolean,
+      default: true,
+    },
+  },
+  setup(props, { slots }) {
+    const popperRef = ref(null as Nullable<HTMLElement>)
+    const arrowRef = ref(null as Nullable<HTMLElement>)
+    const popperInstance = ref(null as PopperInstance)
+    const exceptionState = ref(false)
+    const timeout = ref(null as NodeJS.Timeout)
+    const timeoutPending = ref(null as NodeJS.Timeout)
+    const show = ref(false)
+    const popperId = ref(`el-tooltip-${generateId()}`)
+    const trigger = ref(null as Nullable<HTMLElement>)
+
+    const visible = computed(() => {
+      return props.manualMode ? props.value : !props.disabled && show.value
+    })
+
+    const popperOptions = computed(() => {
+      return {
+        modifierOptions: {
+          arrowOffset: props.arrowOffset,
+          arrowRef: arrowRef.value,
+          boundariesPadding: props.boundariesPadding,
+          cutoff: props.cutoff,
+          offset: props.offset,
+          showArrow: props.showArrow,
+          fallbackOptions: props.flip ? {} : { fallbackPlacements: [] },
+        },
+        placement: props.placement,
+        strategy: props.strategy,
+      }
+    })
+
+    const clearTimer = (timer: Ref<Nullable<NodeJS.Timeout>>) => {
+      clearTimeout(timer.value)
+      timer.value = null
+    }
+    if (!slots.trigger) {
+      throwError('Trigger must be provided')
+    }
+    // this is a reference that we need to pass down to child component
+    // to obtain the child instance
+    function doDestroy(forceDestroy: boolean) {
+      /* istanbul ignore if */
+      if (!popperInstance.value || (visible.value && !forceDestroy)) return
+      detach()
+      if (popperRef.value && props.appendToBody) {
+        off(popperRef.value, 'click', stop)
+        document.body.removeChild(popperRef.value)
+      }
+    }
+
+    function detach() {
+      popperInstance.value.destroy()
+      popperInstance.value = null
+      const _trigger = trigger.value
+      off(_trigger, 'mouseenter', _show)
+      off(_trigger, 'mouseleave', _hide)
+      off(_trigger, 'focus', handleFocus)
+      off(_trigger, 'blur', handleBlur)
+      trigger.value = null
+    }
+
+    const showPopper = () => {
+      if (!exceptionState.value || props.manualMode || props.disabled) return
+      clearTimer(timeout)
+      if (props.openDelay === 0) {
```

Comment:
openDelay is more like the pair of closeDelay rather than hideAfter. showAfter or openAfter maybe better in this context